src/traits/impls.rs

```diff
@@ -53,24 +57,26 @@ impl DoughnutApi for () {
 
 #[cfg(feature = "std")]
 impl Signing for DoughnutV0 {
-    fn sign_ed25519(&mut self, secret_key: &[u8]) -> Result<H512, SigningError> {
+    fn sign_ed25519(&mut self, secret_key: &[u8]) -> Result<Vec<u8>, SigningError> {
         sign_ed25519(&self.issuer(), secret_key, &self.payload())
             .map(|signed_signature| H512::from_slice(&signed_signature))
             .map(|signature| {
-                self.signature = signature;
+                self.signature = signature; // Store signature as type:H512
                 self.signature_version = 1 as u8;
                 signature
             })
+            .map(|signature| H512::to_fixed_bytes(signature).to_vec()) // Export signature as type:Vec<u8>
     }
 
-    fn sign_sr25519(&mut self, secret_key: &[u8]) -> Result<H512, SigningError> {
+    fn sign_sr25519(&mut self, secret_key: &[u8]) -> Result<Vec<u8>, SigningError> {
         sign_sr25519(&self.issuer(), secret_key, &self.payload())
             .map(|signed_signature| H512::from_slice(&signed_signature))
             .map(|signature| {
-                self.signature = signature;
+                self.signature = signature; // Store signature as type:H512
                 self.signature_version = 0 as u8;
```

Comment:
same here, we should set the signature version before signing

Reply:
@alexsednz^

Reply:
Same as above